### ISSUE TITLE
Can't npm install examples

### ISSUE THREAD

**Issue by Jakobud** [gaearon/redux#456]
I get this:

```
[root@dev01 counter]# npm install
npm WARN package.json counter-redux@0.0.0 No README data
npm WARN optional dep failed, continuing fsevents@0.3.8
npm ERR! Linux 3.10.0-123.13.1.el7.x86_64
npm ERR! argv "/usr/bin/node" "/bin/npm" "install"
npm ERR! node v0.12.7
npm ERR! npm  v2.11.3
npm ERR! path /vagrant/repos/redux/examples/counter/node_modules/webpack/node_mo
dules/watchpack/node_modules/chokidar/node_modules/anymatch/node_modules/microma
tch/node_modules/object.omit/node_modules/for-own/node_modules/for-in/package.js
on.e3657e845cd1a5985f9cf49a954965e9
npm ERR! code EPERM
npm ERR! errno -1

npm ERR! Error: EPERM, open '/vagrant/repos/redux/examples/counter/node_modules/
webpack/node_modules/watchpack/node_modules/chokidar/node_modules/anymatch/node_
modules/micromatch/node_modules/object.omit/node_modules/for-own/node_modules/fo
r-in/package.json.e3657e845cd1a5985f9cf49a954965e9'
npm ERR!     at Error (native)
npm ERR!  { [Error: EPERM, open '/vagrant/repos/redux/examples/counter/node_modu
les/webpack/node_modules/watchpack/node_modules/chokidar/node_modules/anymatch/n
ode_modules/micromatch/node_modules/object.omit/node_modules/for-own/node_module
s/for-in/package.json.e3657e845cd1a5985f9cf49a954965e9']
npm ERR!   errno: -1,
npm ERR!   code: 'EPERM',
npm ERR!   path: '/vagrant/repos/redux/examples/counter/node_modules/webpack/nod
e_modules/watchpack/node_modules/chokidar/node_modules/anymatch/node_modules/mic
romatch/node_modules/object.omit/node_modules/for-own/node_modules/for-in/packag
e.json.e3657e845cd1a5985f9cf49a954965e9' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.
npm ERR! Linux 3.10.0-123.13.1.el7.x86_64
npm ERR! argv "/usr/bin/node" "/bin/npm" "install"
npm ERR! node v0.12.7
npm ERR! npm  v2.11.3
npm ERR! path npm-debug.log.0523043cbde7f23d4887250e13c8808c
npm ERR! code ETXTBSY
npm ERR! errno -26

npm ERR! ETXTBSY, rename 'npm-debug.log.0523043cbde7f23d4887250e13c8808c'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /vagrant/repos/redux/examples/counter/npm-debug.log
```

It says to run it as root/Administrator but I'm already doing it as root. Any clues?

**Comment by commandtab** [comment]
@Jakobud `npm install` on the examples is working for me on OS X 10.10.4 with npm 2.12.1 and node 0.12.7.

It looks like you're already running as root inside the Vagrant VM, so perhaps your `/vagrant` directory is a network share that npm is having trouble performing the permissions changes on? If that's the case, are you cable to get Vagrant to run the `npm install` completely inside the virtual machine guest, with files residing on the virtual machine's disk, instead of a shared directory?

**Comment by knowbody** [comment]
you are using vagrant and that is the cause of your problem. try to add:

`config.bindfs.bind_folder "/vagrant-nfs", remote_www_root, u: 'www-data', g: 'vagrant', p: 'u=rwx:g=rwx:o=rD', chown_ignore: true, chmod_ignore: true, m: 'vagrant,www-data'`

to your `Vagrantfile` and you should be fine

**Comment by Jakobud** [comment]
What is `remote_www_root`?

**Comment by knowbody** [comment]
edited, sorry

**Comment by Jakobud** [comment]
Okay this is a problem with Vagrant on a Windows host machine. It is a bug with mounting shared folders with very long (>256 characters) paths. This was a bug they fixed in Vagrant 1.7.3 but reverted in the latest version because of some regression. Thanks guys,
